libnavigation-ui/src/main/java/com/mapbox/navigation/ui/instruction/InstructionView.java

```diff
@@ -239,6 +257,35 @@ public void unsubscribe() {
       navigationViewModel.instructionModel.removeObservers(lifecycleOwner);
       navigationViewModel.bannerInstructionModel.removeObservers(lifecycleOwner);
       navigationViewModel.isOffRoute.removeObservers(lifecycleOwner);
+      navigationViewModel.routeJunctionModel.removeObservers(lifecycleOwner);
+    }
+  }
+
+  ///////////////////////////////////////////////////////////////////
+  public void determineGuidanceView(RouteProgress routeProgress) {
```

Comment:
IMO `determineGuidanceView` is too long.
What about decomposing / extracting blocks of code into private methods / separate classes with single responsibility and make intention clearer? `InstructionView` is getting cluttered 😬